apps/judicial-system/web/src/utils/hooks/useLawyers/useLawyers.ts

```diff
@@ -30,12 +39,24 @@ export const useGetLawyer = (
   nationalId?: string | null,
   shouldFetch?: boolean,
 ): Lawyer | undefined => {
-  const fetchWithNationalId = (url: string, nationalId: string) =>
-    fetch(`${url}?nationalId=${nationalId}`).then((res) => res.json())
+  const { formatMessage } = useIntl()
+
+  const fetcher = (url: string): Promise<Lawyer> =>
+    fetch(url).then((res) => {
+      if (!res.ok) {
+        throw new Error(
+          `Failed to get lawyer with nationalId ${nationalId} from lawyer registry`,
+        )
+      }
 
-  const { data } = useSWR<Lawyer>(
-    nationalId && shouldFetch ? [`/api/lawyers/getLawyer`, nationalId] : null,
-    fetchWithNationalId,
+      return res.json()
+    })
+
+  const { data, error } = useSWR<Lawyer>(
+    nationalId && shouldFetch
+      ? `/api/defender/lawyerRegistry/${nationalId}`
```

Comment:
Nit: Should we extract `/api/defender/lawyerRegistry` since its used in 2x paces in this hook file?

Reply:
Sures! Done 👍